samples/README.md

```diff
@@ -0,0 +1,27 @@
+# Samples, Snippets, and How-To Guides
+
+Developers like to learn by looking at code, and so the Azure SDK comes with many samples in the form of short code snippets, sample applications, and how-to guides. This document describes where to find all these resources.
+
+## Structure of the Repository
+The Azure SDK repository is organized in the following folder structure, with the main sample locations highlighted using **bold** font.
+
+- `/samples` (this folder)
+  - `README.md` (this file)
+- `/sdk` (folder containing sources for all SDK packages)
+  - `/<service>` (e.g. storage)
+    - `/<package>` (e.g. `@azure/storage-blob`)
+      - **`README.md`** (package READMEs contain "hello world" code snippets)
+      - **`/samples`**
+        - **`/typescript`** (package-specific samples in TypeScript)
+        - **`/javascript`** (package-specific samples in JavaScript using CommonJS)
+      - `/src`
+      - `/tests`
+
+##  Getting Started (a.k.a. `Hello World`) Samples
+Each package folder contains a package-specific `README.md` file. Most of these `README` files contain `Hello World` code samples illustrating basic usage of the the APIs contained in the package. For example, you can find `Hello World` samples for the `@azure/storage-blob` package [here](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/storage/storage-blob#examples).
```

Comment:
Hm do we use the term `Hello World` in other docs/references? I think it is a good cultural touchstone for programmers, but not sure if we use it as a regular term or not

Reply:
.NET uses this term in its corresponding file. I'm not attached to it... we're considerably past the bare basics of the APIs in many of the readmes.

Personally -- I think that a lot of what's in our readmes is redundant with what's in the samples files, but that's a different question entirely. Maybe we could just call them "introductory samples" or something like that. Just "Getting Started Samples" would also work.

Reply:
I have a small preference for "getting started code" or somesuch